### PR TITLE
feat(sdk): add JobsService list, exists, restart, stop 

### PR DESCRIPTION
**Pull request by uipreliga**
This pull request introduces a shared validation utility for pagination parameters, refactors existing services to use it, and adds comprehensive tests to ensure consistent pagination and bulk job operations. The changes improve code reuse, maintainability, and reliability of pagination and job resolution logic across the SDK.

**Shared validation utility:**

* Added a new `validate_pagination_params` function in `src/uipath/_utils/validation.py` to centralize and standardize pagination parameter checks for SDK services.
* Exposed `validate_pagination_params` in the public API via `src/uipath/_utils/__init__.py`. [[1]](diffhunk://#diff-2c12ba5229a779492ade3e011968e3d74c0b55d4dd79e1b9929eeb812fe542bfR8) [[2]](diffhunk://#diff-2c12ba5229a779492ade3e011968e3d74c0b55d4dd79e1b9929eeb812fe542bfR20)

**Service refactoring for pagination validation:**

* Refactored `BucketsService.list` and `BucketsService.list_async` methods to use the shared `validate_pagination_params` utility instead of inline checks, ensuring consistent validation logic. [[1]](diffhunk://#diff-aef83095f3372a97f8f8345e48bd95f004f534d234c3deafb999af8878a2a906R14) [[2]](diffhunk://#diff-aef83095f3372a97f8f8345e48bd95f004f534d234c3deafb999af8878a2a906L98-R104) [[3]](diffhunk://#diff-aef83095f3372a97f8f8345e48bd95f004f534d234c3deafb999af8878a2a906L182-R180)

**Testing improvements:**

* Added extensive tests for the new validation utility in `tests/sdk/test_utils_validation.py`, covering edge cases and boundary values.
* Added a fixture for `JobsService` to support new and existing tests in `tests/sdk/services/conftest.py`.

**JobsService enhancements and testing:**

* Added tests for bulk job operations and N+1 query fix in `tests/sdk/services/test_jobs_service_bulk_operations.py`, ensuring efficient job resolution and correct API usage.
* Added tests for pagination behavior and validation in `JobsService` with `tests/sdk/services/test_jobs_service_pagination.py`, verifying correct paged results and validation error handling.

### CODE REVIEW

File: src/uipath/_services/jobs_service.py [hunk 1]
```diff
@@ -920,3 +1260,221 @@ async def main():
...
+        self,
+        job_id: int,
+        folder_key: Optional[str],
+        folder_path: Optional[str],
+    ) -> RequestSpec:
+        """Build request for restarting a job."""
+        return RequestSpec(
+            method="POST",
+            endpoint=Endpoint(
+                "/orchestrator_/odata/Jobs/UiPath.Server.Configuration.OData.RestartJob"
```
Comment (busesorin94) [inline src/uipath/_services/jobs_service.py]: Restart has a weird behavior because it doesn't start on the same machine and identity. You can rely on it only to start the process

File: src/uipath/_services/jobs_service.py [hunk 2]
```diff
@@ -148,6 +155,324 @@ async def main():  # noqa: D103
...
+    async def stop_async(
+        self,
+        *,
+        job_keys: List[str],
+        strategy: str = "SoftStop",
+        folder_path: Optional[str] = None,
+        folder_key: Optional[str] = None,
+    ) -> None:
+        """Async version of stop()."""
+        job_ids = await self._resolve_job_identifiers_async(
```
[the remark this answers is not in the document]
Comment (uipreliga) [reply src/uipath/_services/jobs_service.py]: We're already using /odata/Jobs/...StopJobs with jobIds
Is this what you intended, or did you have something else in mind?

File: src/uipath/_services/jobs_service.py (hunk 1 above)
Comment (uipreliga) [reply src/uipath/_services/jobs_service.py]: Not sure i understand - should i remove the restart command or change something else?
